### PR TITLE
[various] update agp and gradle for all examples in packages

### PR DESCRIPTION
**Pull request by reidbaker**
https://github.com/flutter/flutter/issues/122213
Update gradle and AGP for all examples in packages

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]

**Review by tarrinneal — APPROVED** [review]
image_picker, pigeon, and video_player lgtm

**Comment by stuartmorgan** [comment]
> Repo checks is failing because of image_picker not having a changelog but it appears all modified files are in example directories.

That's WAI; changes to example files that aren't on a specific exclusion list (like build.gradle) aren't automatically ignored for the check, so that a human remembers to look and see if it's worth calling out in the CHANGELOG. E.g., if the change to the manifest (which is what it's noticing I believe) were, say, adding a `uses-permission` because someone noticed that our example was not showing us requesting a permission that plugin client apps should have, we might want to flag that in the CHANGELOG as something like "Updates example app to show that permission SOME_IMPORTANT_PERMISSION is required."

In this case the change to the manifest is uninteresting, so we can just override it the check now that we've looked at it and made a deliberate choice.

**Review by chunhtai — APPROVED** [review]
LGTM, go_router

File: packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/TileProviderController.java [hunk 1]
```diff
@@ -84,9 +84,12 @@ public void success(Object data) {
     public void error(String errorCode, String errorMessage, Object data) {
       Log.e(
```
Comment (reidbaker) [inline packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/TileProviderController.java:85]: I want to call out that this log message is incorrect It uses errorCode for code and message. That said I am not modifying that logic in this PR only reformatting the string format b/c the linter had an issue.

**Comment by reidbaker** [comment]
Holding to land after https://github.com/flutter/packages/pull/3836 which is more urgent.

**Review by camsim99 — APPROVED** [review]
Changes lgtm!

**Review by gmackall — APPROVED** [review]

**Comment by stuartmorgan** [comment]
Sorry about all the conflicts 🙁

**Pull request merged**